docs/getting-started/actions.md

```diff
@@ -2,73 +2,143 @@
 title: Actions
 ---
 
-You can update Variables through Dara actions which can be passed to any component prop accepting an `Action`.
+Actions are a way to add interactivity to your Dara app in response to some user interaction. Components accept actions as arguments to their callbacks annotated with type `Action`, usually named `onclick`, `onchange` or similar. Actions can be used to update variables, trigger notifications, navigate to a different page, and more.
 
-## `UpdateVariable`
+A valid argument to an action parameter is one of the following three options:
 
-The `dara.core.interactivity.actions.UpdateVariable` action can trigger the update of a `Variable`, `DataVariable` or `UrlVariable`.
+- an instance of an `@action`-decorated function (as in the example above)
+- an instance of an individual static action object
+- a list of a combination of the above two options
 
-It takes the following arguments:
+In this section we will go through all the different ways to create actions and how to use them.
 
-- `resolver`: a function to resolve the new value for the `Variable`, `DataVariable` or `UrlVariable`. The resolver takes one argument: a context of type `UpdateVariable.Ctx`. The new value is given by the component and can be obtained with `ctx.inputs.new`. While the current value, or now previous value of the variable you are updating, can be obtained with `ctx.inputs.old`.
-- `variable`: the `Variable`, `DataVariable` or `UrlVariable` to update with a new value upon triggering the action
-- `extras`: any extra variables to resolve and pass to the resolver function, you can obtain a list of the resolved values of all the extras passed with `ctx.extras` in the resolver function
+## `@action` decorator
+
+The primary way to create actions in Dara is the `@action` decorator. This decorator takes a function and returns an action that can be passed to a component's callback.
+It injects an `ActionCtx` object as the first argument of the function, which contains the input sent from the component and exposes action methods to update the state of the application.
+
+:::note
+
+The action methods on `ActionCtx` are `async` and must be `await`-ed, so your decorated function should be `async` as well.
 
-To see how the `UpdateVariable` works, consider the example below:
+:::
 
 ```python
-from dara.core import ConfigurationBuilder, Variable, UpdateVariable
+from dara.core import action, Variable
```

Comment:
Can we try to make this example work end to end so that it allows the reader to just c&p and run? Can be a simple example which demonstrates the main point

Reply:
sorry, missread